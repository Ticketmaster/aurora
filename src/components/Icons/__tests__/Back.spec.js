/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";

import { BackIcon } from "../";

describe("BackIcon", () => {
  const children = <title>Back Icon</title>;

  it("renders default correctly", () => {
    const component = renderer.create(<BackIcon>{children}</BackIcon>);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders large correctly", () => {
    const component = renderer.create(
      <BackIcon size="large">{children}</BackIcon>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
