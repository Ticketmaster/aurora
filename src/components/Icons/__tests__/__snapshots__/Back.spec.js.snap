// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackIcon renders default correctly 1`] = `
<svg
  height="16"
  viewBox="0 0 16 16"
  width="16"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Back Icon
  </title>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <path
      d="M15.8.133h-16v16h16z"
    />
    <path
      d="M10.32 12.857a.75.75 0 0 1-1.062 1.057l-5.24-5.263a.75.75 0 0 1 0-1.059L9.26 2.354a.75.75 0 0 1 1.06 1.06l-4.711 4.71 4.712 4.733z"
      fill="#262626"
      fillOpacity=".65"
      fillRule="nonzero"
    />
  </g>
</svg>
`;

exports[`BackIcon renders large correctly 1`] = `
<svg
  height="24"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Back Icon
  </title>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <path
      d="M24 0H0v24h24z"
    />
    <path
      d="M10.773 20.738a.75.75 0 0 1-1.106 1.013L1.197 12.5a.75.75 0 0 1 0-1.013l8.47-9.243a.75.75 0 1 1 1.106 1.014l-8.006 8.736 8.006 8.745z"
      fill="#262626"
      fillOpacity=".65"
      fillRule="nonzero"
    />
  </g>
</svg>
`;

exports[`BackIcon renders xSmall correctly 1`] = `
<svg
  height="12"
  viewBox="0 0 12 12"
  width="12"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Back Icon
  </title>
  <g
    fill="none"
    fillRule="evenodd"
    transform="translate(-.2 -.867)"
  >
    <rect
      height="12"
      rx="1"
      width="12"
      y="1"
    />
    <path
      d="M8.301 10.453a.75.75 0 1 1-1.056 1.065L3.222 7.533a.75.75 0 0 1 0-1.066l4.01-3.972A.75.75 0 0 1 8.288 3.56L4.816 7 8.3 10.453z"
      fill="#262626"
      fillOpacity=".65"
      fillRule="nonzero"
    />
  </g>
</svg>
`;
